### PR TITLE
feat(groups): groups filter edit and update for hook

### PR DESCRIPTION
**Pull request by Fewwy**
The point of this PR is to make groups data available inside the entities.
That should allow us to get the same data in the consumer applications without any hard-coded changes.

How to test:
run local dev env using this branch
1)Check the functionality of the groups actions in inventory table
2)Check the functionality of the groups table
3)Check the single group page and it's functionality.

Check the redux store - the entities should have a "groups" object that is filled with data

Had to refactor the useEffect, and pull out the async function out of it. Otherwise it was breaking the groups table and causing this error https://medium.com/geekculture/react-uncaught-typeerror-destroy-is-not-a-function-192738a6e79b

### CODE REVIEW

File: src/store/entities.js [hunk 1]
```diff
@@ -210,6 +210,16 @@ function changeSort(state, { payload: { key, direction } }) {
     };
 }
 
+function groupsLoaded(state, { payload }) {
+    return {
+        ...state,
+        groups: payload
+        /* Promise.resolve(groupsApi.apiGroupGetGroupList(undefined, 50, undefined, 'name', 'ASC', undefined))
```
[the remark this answers is not in the document]
Comment (Fewwy) [reply src/store/entities.js:217]: Yes, thanks :)

File: src/store/entities.js [hunk 2]
```diff
@@ -52,7 +52,7 @@ export const defaultColumns = (groupsEnabled = false) => ([
         title: 'Group',
         props: { width: 10, isStatic: true },
         // eslint-disable-next-line camelcase
-        renderFunc: (groups) => isEmpty(groups) ? 'N/A' : groups[0].name, // currently, one group at maximum is supported
+        renderFunc: (groups) => isEmpty(groups) ? 'N/A' : groups[0]?.name, // currently, one group at maximum is supported
```
Comment (gkarat) [inline src/store/entities.js:55]: Is this change really needed? If the groups are empty, then 'N/A' will be rendered. Otherwise, we expect that groups contain the data with valid format (here means, containing `name`).

File: cypress/fixtures/hosts.json [hunk 3]
```diff
@@ -2389,7 +2389,347 @@
       "stale_warning_timestamp": "1958-08-14T23:00:00.0Z",
       "stale_timestamp": null,
       "mac_addresses": null,
-      "groups": []
+      "groups": {
+          "total": 37,
+          "count": 37,
+          "page": 1,
+          "per_page": 50,
+          "results": [
```
Comment (gkarat) [inline cypress/fixtures/hosts.json:2397]: `groups` is a list of groups that a particular host has. You must probably copy-pasted response from /groups which is not correct here.

**Review by gkarat — CHANGES_REQUESTED** [review]

File: src/store/entities.js (hunk 2 above)
Comment (Fewwy) [reply src/store/entities.js:55]: Agree, removed the conditional chaining

File: src/store/inventory-actions.js [hunk 4]
```diff
@@ -216,3 +217,8 @@ export const toggleDrawer = (isOpened) => ({
     type: TOGGLE_DRAWER,
     payload: { isOpened }
 });
+
+export const fetchGroupsForEntities = () => ({
```
[the remark this answers is not in the document]
Comment (Fewwy) [reply src/store/inventory-actions.js]: Done

File: cypress/fixtures/hosts.json (hunk 3 above)
Comment (Fewwy) [reply cypress/fixtures/hosts.json]: Removed the meta data from the groups, removed the results property.
Now the groups is just an array of objects

File: src/components/InventoryTable/EntityTableToolbar.test.js [hunk 5]
```diff
@@ -330,8 +330,8 @@ describe('EntityTableToolbar', () => {
                 </Provider>);
                 wrapper.find('.ins-c-chip-filters button.pf-m-link').last().simulate('click');
                 const actions = store.getActions();
-                expect(actions.length).toBe(3);
-                expect(actions[actions.length - 2]).toMatchObject({ type: 'CLEAR_FILTERS' });
+                expect(actions.length).toBe(4);
+                expect(actions[actions.length - 3]).toMatchObject({ type: 'CLEAR_FILTERS' });
```
[the remark this answers is not in the document]
Comment (Fewwy) [reply src/components/InventoryTable/EntityTableToolbar.test.js:334]: I added a check for that

**Review by mkholjuraev — APPROVED** [review]
LGTM!

**Review by gkarat — APPROVED** [review]
